src/main/java/htsjdk/samtools/SamReaderFactory.java

```diff
@@ -323,7 +325,25 @@ public SamReader open(final SamInputResource resource) {
                     return reader;
                   }
                 }
-                if (type == InputResource.Type.SEEKABLE_STREAM || type == InputResource.Type.URL) {
+                if (type == InputResource.Type.URI) {
+                    final URI uri = ((UriInputResource) data).uri;
+                    if (uri.getScheme().equalsIgnoreCase("htsget")) {
+                        final URI httpUri = new URI(
```

Comment:
I would extract out a function `HtsgetBAMFileReader.convertHtsgetUriToHttps` to document what this is doing.